packages/grafana-ui/src/components/Layout/Box/Box.tsx

```diff
@@ -0,0 +1,267 @@
+  alignItems?: ResponsiveProp<AlignItems>;
+  justifyContent?: ResponsiveProp<JustifyContent>;
+
+  // Other props
+  backgroundColor?: ResponsiveProp<BackgroundColor>;
+  display?: ResponsiveProp<Display>;
+  boxShadow?: ResponsiveProp<BoxShadow>;
+  /** Sets the HTML element that will be rendered as a Box. Defaults to 'div' */
+  element?: ElementType;
+}
+
+export const Box = ({
+  children,
+  margin,
+  marginX,
+  marginY,
+  marginTop,
+  marginBottom,
+  marginLeft,
+  marginRight,
+  padding,
+  paddingX,
+  paddingY,
+  paddingTop,
+  paddingBottom,
+  paddingLeft,
+  paddingRight,
+  display,
+  backgroundColor,
+  grow,
+  shrink,
+  borderColor,
+  borderStyle,
+  borderRadius,
+  justifyContent,
+  alignItems,
+  boxShadow,
+  element,
+}: React.PropsWithChildren<BoxProps>) => {
+  const styles = useStyles2(
+    getStyles,
+    margin,
+    marginX,
+    marginY,
+    marginTop,
+    marginBottom,
+    marginLeft,
+    marginRight,
+    padding,
+    paddingX,
+    paddingY,
+    paddingTop,
+    paddingBottom,
+    paddingLeft,
+    paddingRight,
+    display,
+    backgroundColor,
+    grow,
+    shrink,
+    borderColor,
+    borderStyle,
+    borderRadius,
+    justifyContent,
+    alignItems,
+    boxShadow
+  );
+  const Element = element ?? 'div';
+
+  return <Element className={styles.root}>{children}</Element>;
+};
+
+Box.displayName = 'Box';
+
+const customBorderColor = (color: BorderColor, theme: GrafanaTheme2) => {
+  switch (color) {
+    case 'error':
+    case 'success':
+    case 'info':
+    case 'warning':
+      return theme.colors[color].border;
+    default:
+      return color ? theme.colors.border[color] : undefined;
+  }
+};
+
+const customBackgroundColor = (color: BackgroundColor, theme: GrafanaTheme2) => {
+  switch (color) {
+    case 'error':
+    case 'success':
+    case 'info':
+    case 'warning':
+      return theme.colors[color].transparent;
+    default:
+      return color ? theme.colors.background[color] : undefined;
+  }
+};
+
+const getStyles = (
+  theme: GrafanaTheme2,
+  margin: BoxProps['margin'],
+  marginX: BoxProps['marginX'],
+  marginY: BoxProps['marginY'],
+  marginTop: BoxProps['marginTop'],
+  marginBottom: BoxProps['marginBottom'],
+  marginLeft: BoxProps['marginLeft'],
+  marginRight: BoxProps['marginRight'],
+  padding: BoxProps['padding'],
+  paddingX: BoxProps['paddingX'],
+  paddingY: BoxProps['paddingY'],
+  paddingTop: BoxProps['paddingTop'],
+  paddingBottom: BoxProps['paddingBottom'],
+  paddingLeft: BoxProps['paddingLeft'],
+  paddingRight: BoxProps['paddingRight'],
+  display: BoxProps['display'],
+  backgroundColor: BoxProps['backgroundColor'],
+  grow: BoxProps['grow'],
+  shrink: BoxProps['shrink'],
+  borderColor: BoxProps['borderColor'],
+  borderStyle: BoxProps['borderStyle'],
+  borderRadius: BoxProps['borderRadius'],
+  justifyContent: BoxProps['justifyContent'],
+  alignItems: BoxProps['alignItems'],
+  boxShadow: BoxProps['boxShadow']
+) => {
```

Comment:
```suggestion
type StyleProps = Omit<BoxProps, 'children' | 'element'>;
const getStyles = (
  theme: GrafanaTheme2,
  {
    margin,
    marginX,
    marginY,
    marginTop,
    marginBottom,
    marginLeft,
    marginRight,
    padding,
    paddingX,
    paddingY,
    paddingTop,
    paddingBottom,
    paddingLeft,
    paddingRight,
    display,
    backgroundColor,
    grow,
    shrink,
    borderColor,
    borderStyle,
    borderRadius,
    justifyContent,
    alignItems,
    boxShadow,
  }: StyleProps
) => {
```
I think we could reuse the component's prop types here. This would allow simplifying the actual component like so: 
```
export const Box = ({ element, children, ...props }: React.PropsWithChildren<BoxProps>) => {
  const styles = useStyles2(getStyles, props);
  const Element = element ?? 'div';

  return <Element className={styles.root}>{children}</Element>;
};
```

Reply:
I tried this before, but it seems like having just `prop` in that `useStyles2` does not memo it correctly and it will call `getStyles` multiple times if there are `Boxes` with the same props in a page 😞 

https://github.com/grafana/grafana/pull/73637#discussion_r1331499837